ms2deepscore/MS2DeepScoreData.py

```diff
@@ -0,0 +1,166 @@
+from tqdm import tqdm
+from ms2deepscore.spectrum_binning_fixed import create_peak_list_fixed
+from ms2deepscore.spectrum_binning_fixed import set_d_bins_fixed
+from ms2deepscore.spectrum_binning_fixed import unique_peaks_fixed
+from ms2deepscore.utils import create_peak_dict
+
+
+class MS2DeepScoreData:
+    """Create MS2DeepScore binned spectrum data and keep track of parameters.
+
+    TODO: add description --> here: fixed bins!
+    """
+    def __init__(self, number_of_bins: int,
+                 mz_max: float = 1000.0, mz_min: float = 10.0,
+                 peak_scaling: float = 0.5, allowed_missing_percentage: float = 0.0):
+        """
+
+        Parameters
+        ----------
+        number_of_bins
+            Number if bins to represent spectrum.
+        mz_max
+            Upper bound of m/z to include in binned spectrum. Default is 1000.0.
+        mz_min
+            Lower bound of m/z to include in binned spectrum. Default is 10.0.
+        peak_scaling
+            Scale all peak intensities by power pf peak_scaling. Default is 0.5.
+        allowed_missing_percentage:
+            Set the maximum allowed percentage of the spectrum that may be unknown
+            from the input model. This is measured as percentage of the weighted, unknown
+            binned peaks compared to all peaks of the spectrum. Default is 0, which
+            means no unknown binned peaks are allowed.
+        """
+        self.number_of_bins = number_of_bins
+        assert mz_max > mz_min, "mz_max must be > mz>min"
+        self.mz_max = mz_max
+        self.mz_min = mz_min
+        self.d_bins = set_d_bins_fixed(number_of_bins, mz_min=mz_min, mz_max=mz_max)
+        self.peak_scaling = peak_scaling
+        self.allowed_missing_percentage = allowed_missing_percentage
+        self.peak_to_position = None
+        self.known_bins = None
+        self.generator_args = {}
+        self.spectrums_binned = None
+        self.inchikeys_all = None
+
+    def collect_binned_spectrums(self, spectrums: list, progress_bar=True):
+        """Create 'vocabulary' of bins that have peaks in spectrums.
+        Derive binned spectrums from spectrums and store them.
+
+        Parameters
+        ----------
+        spectrums
+            List of spectrums.
+        progress_bar
+            Show progress bar if set to True. Default is True.
+        """
+        print("Collect spectrum peaks...")
+        peak_to_position, known_bins = unique_peaks_fixed(spectrums, self.d_bins, self.mz_min)
+        print(f"Calculated embedding dimension of {len(known_bins)}.")
+        self.peak_to_position = peak_to_position
+        self.known_bins = known_bins
+
+        print("Convert spectrums to binned spectrums...")
+        self.spectrums_binned = self.create_binned_spectrums(spectrums, progress_bar)
+
+        # Collect inchikeys
+        self._collect_inchikeys(spectrums)
+
+    def create_binned_spectrums(self, input_spectrums: list, progress_bar=True):
```

Comment:
I would set `progress_bar` in the constructor of this class instead of in this method.